src/interpreter/kernel/exec.js

```diff
@@ -747,6 +772,83 @@ export function executeStackFrame(
         break;
       }
 
+      // https://webassembly.github.io/spec/core/exec/instructions.html#and
+      case "load":
+      case "load16_s":
+      case "load16_u":
+      case "load8_s":
+      case "load8_u":
+      case "load32_s":
+      case "load32_u": {
+        const memory = getMemory();
+
+        const ptr = pop1("i32").value.toNumber() + getMemoryOffset(instruction);
```

Comment:
Just an idea but the `pop{1,2}` is performing a validation from the spec, should we rename to `validateAndPop{1,2}` to make it clear? (don't need to do this in this pr).

Reply:
I think most of the helper methods within `exec` will perform some validation, so I wouldn't bother renaming them :-)